pkg/controller/vsphere/session/tag_ids_caching_client.go

```diff
@@ -0,0 +1,242 @@
+package session
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi/vapi/tags"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	defaultCacheTTL = time.Hour * 12
+
+	// special value for determine whenever tag or category was found in vCenter during last lookup
+	notFoundValue      = "NOT_FOUND"
+	notFoundErrMessage = "404 Not Found"
+)
+
+type cachedItem struct {
+	value     string
+	expiresAt int64
+}
+
+func (ci cachedItem) expired() bool {
+	return time.Now().UnixNano() > ci.expiresAt
+}
+
+type cacheMap struct {
+	internalMap sync.Map
+}
+
+// Set value by the given key with default 12 hours lifetime
+func (c *cacheMap) Set(key string, value string) {
+	c.internalMap.Store(key, cachedItem{value: value, expiresAt: time.Now().Add(defaultCacheTTL).UnixNano()})
```

Comment:
Why doesn't this call `SetWithTTL`? It's basically the same right but with a default TTL? Could clean this up a bit